### PR TITLE
fix(pages) fix inconsistency in items in SSR and Clientside rendering

### LINKED ISSUE #3138: Docs(configuration): Performance>Items link to Node>Items in Sidebar/SidebarItem
Hello brave webpack community,

There are falsy links to configuration items in the Sidebar in the current configuration docs for webpack 4.35.0 under https://webpack.js.org/configuration/

The all the "Performance" SidebarItem's actual link to the "Node" doc items.

As an example:
`Performance > Hints` in Sidebar links to https://webpack.js.org/configuration/node/#nodeconsole whereas it should link to https://webpack.js.org/configuration/performance/#performancehints)
The following screenshot demonstrates exactly that:
(Performance > Hints link in Sidebar appended):

I was trying to track this down in this repo, but didn't find the issue neither in the Markdown files nor in your documentation build chain.

Maybe one of you who is native to the docs generating is doing a better job than me :/

### PR DESCRIPTION
**Pull request by EugeneHlushko**
TLDR; Fixes #3138 

Originally i thought the issue was caused by something we'd missed in printable feature. But later i realized it was inconsistency in sort in between SSG and Clientside runs specifically React.dom's `hydrate` method. After reading about similar issues it was clear we had two options. Make sure elements orders are consistent or ditch using `hydrate`. I gave a shot to a good sort and it worked with `hydrate`.

Related information: https://github.com/facebook/react/issues/10591#issuecomment-333174218

### CODE REVIEW

**Comment by EugeneHlushko** [comment]
This is safe to merge and also kind of critical, broken urls never good. @montogeek

**Comment by montogeek** [comment]
I would rather understand the root issue here, Why adding a new element to the sidebars create this inconsistency that makes us change all the order in the articles?

**Comment by EugeneHlushko** [comment]
Its not that element, see reference issue

**Comment by EugeneHlushko** [comment]
The same sort order makes it ordered randomly on server and in browsers e.g. sort [4,4] and if they are randomly sorted in array ssr hydrate can fail, they admit it in the linked comment

**Comment by EugeneHlushko** [comment]
```
var pages = [{title: 'printable'}, {title: 'petyua', sort: 4}, {title: 'vasa', sort: 4}, {title: 'printable', sort: 15}, {title: 'volodya', sort: 16}];
console.log(pages.sort((pageA, pageB) => pageA.sort > pageB.sort));

var pages2 = [{title: 'vasa', sort: 4}, {title: 'printable', sort: 15}, {title: 'petyua', sort: 4}, {title: 'volodya', sort: 16},];
console.log(pages2.sort((pageA, pageB) => pageA.sort > pageB.sort));
```

**Comment by montogeek** [comment]
Hope this fixes it, just navigate through the preview, seems fine.

**Comment by montogeek** [comment]
Thanks!

**Comment by EugeneHlushko** [comment]
By the way, to detect the brokenness you have to reload each page so that `hydrate` fires onto SSG page. At first it wasnt obvious to me why it happened until i stumbled upon the ref issue.

**Comment by EugeneHlushko** [comment]
Because if you navigate clientside from the landing page (one where bundle applied hydrate onto), consequent pageviews look ok
